inc/command/class-command.php

```diff
@@ -215,6 +439,167 @@ protected function run_command( InputInterface $input, OutputInterface $output,
+				],
+			] ), $output );
+		}
+
+		return $return_val;
+	}
+
+	/**
+	 * Create test databases.
+	 *
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 *
+	 * @return int
+	 */
+	protected function delete_test_db( InputInterface $input, OutputInterface $output ) {
+		$use_chassis = $input->getOption( 'chassis' );
+
+		if ( $use_chassis ) {
+			$cli = $this->getApplication()->find( 'chassis' );
+
+			$return_val = $cli->run( new ArrayInput( [
+				'subcommand' => 'exec',
+				'options' => [
+					'mysql',
+					'-uroot',
+					'-ppassword',
+					'-e',
+					'"DROP DATABASE test; DROP DATABASE test2; REVOKE ALL PRIVILEGES on test.* FROM wordpress; REVOKE ALL PRIVILEGES on test2.* FROM wordpress;"',
+				],
+			] ), $output );
+		} else {
+			$cli = $this->getApplication()->find( 'local-server' );
+
+			$return_val = $cli->run( new ArrayInput( [
+				'subcommand' => 'db',
+				'options' => [
+					'exec',
+					'DROP DATABASE test; DROP DATABASE test2; REVOKE ALL PRIVILEGES on test.* FROM wordpress; REVOKE ALL PRIVILEGES on test2.* FROM wordpress;',
+				],
+			] ), $output );
+		}
+
+		return $return_val;
+	}
+
+	/**
+	 * Get the name of the project for the local subdomain
+	 *
+	 * @return string
+	 */
+	protected function get_project_subdomain() : string {
+
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		$composer_json = json_decode( file_get_contents( getcwd() . '/composer.json' ), true );
+
+		if ( isset( $composer_json['extra']['altis']['modules']['local-server']['name'] ) ) {
+			$project_name = $composer_json['extra']['altis']['modules']['local-server']['name'];
+		} else {
+			$project_name = basename( getcwd() );
+		}
+
+		return preg_replace( '/[^A-Za-z0-9\-\_]/', '', $project_name );
+	}
+
+	/**
+	 * Start a headless browser container for WebDriver used in acceptance tests.
+	 *
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 *
+	 * @return int
+	 */
+	protected function start_browser_container( InputInterface $input, OutputInterface $output ) {
+		$columns = exec( 'tput cols' );
+		$lines = exec( 'tput lines' );
+		$browser = $input->getOption( 'browser' );
+
+		$available_browsers = [
+			'chrome',
+			'firefox',
+			// 'edge', // TODO Buggy driver, dig deeper
+		];
+
+		if ( ! in_array( $browser, $available_browsers, true ) ) {
+			throw new CommandNotFoundException( sprintf(
+				'Browser "%s" is unavailable, available browsers are: %s.',
+				$browser,
+				implode( ', ', $available_browsers ),
+			) );
+		}
+
+		$base_command = sprintf(
+			'docker run ' .
+				'-d ' .
+				'-e COLUMNS=%1%d -e LINES=%2$d ' .
+				// '-p 4444:4444 ' .
+				// '-p 7900:7900 ' .
+				'--network=host ' .
+				'--name=%3$s_selenium ' .
+				'--shm-size="2g" ' .
+				'selenium/standalone-%4$s:4.0.0-20211102',
+			$columns,
+			$lines,
+			$this->get_project_subdomain(),
+			$browser,
+		);
+
+		passthru( $base_command, $return_var );
```

Comment:
Really weird but when I ran a browser test it failed on me, I added a `sleep( 10 )` to see if it's because the container wasn't fully up and that got things working. Probably doesn't need 10 seconds though.

Reply:
Hmmm, interesting.. I did not get that.. do you think we should add it anyways ?

Reply:
Yeah, if it can happen at all let's try with 3 seconds for now and adjust after more testing.

```suggestion
		passthru( $base_command, $return_var );

		// Allow time for selenium app to boot up.
		sleep( 3 );
```